services/MerchStoreService.ts

```diff
@@ -100,47 +100,70 @@ export default class MerchStoreService {
     const item = await this.transactions.readOnly(async (txn) => Repositories
       .merchStoreItem(txn)
       .findByUuid(uuid));
-    if (!item) throw new NotFoundError('Item not found');
+    if (!item) throw new NotFoundError('Merch item not found');
     return item.getPublicMerchItem();
   }
 
   public async createItem(item: MerchItem): Promise<MerchandiseItemModel> {
     return this.transactions.readWrite(async (txn) => {
+      item.hasVariantsEnabled ??= false;
+      this.verifyVariantAndOptionConsistency(item);
+
       const collection = await Repositories.merchStoreCollection(txn).findByUuid(item.collection);
-      if (!collection) throw new NotFoundError('Collection not found');
+      if (!collection) throw new NotFoundError('Merch collection not found');
+
       const merchItemRepository = Repositories.merchStoreItem(txn);
       const merchItem = MerchandiseItemModel.create({ ...item, collection });
       await merchItemRepository.upsertMerchItem(merchItem);
       return merchItemRepository.findByUuid(merchItem.uuid);
     });
   }
 
+  // verify that item.hasVariantsEnabled is consistent with the item options length and option type consistency
+  // as defined by the merch store functionality
+  private verifyVariantAndOptionConsistency(item: MerchItem | MerchandiseItemModel) {
+    if (!item.hasVariantsEnabled && item.options.length > 1) {
+      throw new UserError('Items with variants disabled cannot have multiple options');
+    }
+    if (item.hasVariantsEnabled && this.hasMultipleOptionTypes(item.options)) {
+      throw new UserError('Item cannot have multiple option types');
```

Comment:
To match the other error messages.
```suggestion
      throw new UserError('Merch items with variants disabled cannot have multiple options');
    }
    if (item.hasVariantsEnabled && this.hasMultipleOptionTypes(item.options)) {
      throw new UserError('Merch items cannot have multiple option types');
```